torch/_inductor/lowering.py

```diff
@@ -4790,7 +4790,9 @@ def allreduce(input, reduce_op, tag, ranks, group_size):
     @register_lowering(c10d_functional.all_gather_into_tensor)
     def all_gather_into_tensor(shard, tag, ranks, group_size):
         return TensorBox.create(
-            ir.AllGatherIntoTensor.create(shard, tag, ranks, group_size)
+            ir.AllGatherIntoTensor.create(
+                ir.ExternKernel.require_contiguous(shard), tag, ranks, group_size
```

Comment:
is this a pure fix or an optimization/tradeoff?  Guess i'm not sure what require_contiguous does exactly. 

Is the issue that previously the input to allgather may have been accessed incorrectly by allgather due to ignoring actual strides?

Reply:
This is a pure fix.